src/ActionModal/utils/MessageConstructor.js

```diff
@@ -0,0 +1,48 @@
+export const getMessage = async (
+  messageType,
+  transactionProperties,
+  context
+) => {
+  const messageConstructor = await getMessageConstructor(context)
+  const message = messageConstructor(
+    messageType,
+    context.userAddress,
+    transactionProperties
+  )
+  return message
+}
+
+const getMessageConstructor = async context => {
```

Comment:
just out of curiosity - is this going to scale to other signing schemes?

Reply:
the idea was to use this approach to scale yea ^^'